doc/design/bootstrap.md

```diff
@@ -0,0 +1,78 @@
+# Bootstrap Design
+
+etcd currently uses SRV records to bootstrap while this is effective managing an external source of truth is not optimal. This design allows for any source of truth to be used containing a membership list.
+
+## General design
+
+Bootstrap assumes that no existing cluster exists. The bootstrap mechanism can only be used during the bootstrap phase of cluster lifecycle. The design has 2 phases `initialization` and `population`.
```

Comment:
From your non goals I see that by "no existing cluster exists" you mean "we're creating a brand new cluster", is there any possibility or value in trying to use this codepath for the "no cluster is running right now" case in the future too?

Reply:
I will clarify, the example
Is populating snapshot.RestoreConfig{} which is provided by clientv3. So any restore would use that. To your point we could use the same functionality of clientv3 for a full restore but not reconfiguration of a running cluster. I will change that.